### PR TITLE
Add Debuggable properties to Abort

### LINKED ISSUE #1804: Impossible to set Abort properties
It's impossible to set some of the `Abort` struct attributes.

- possibleCauses
- suggestedFixes
- documentationLinks
- stackOverflowQuestions
- gitHubIssues

They are inherited from `Debuggable`, but they are not defined in `Abort.init` nor as `var`.

### PR DESCRIPTION
**Pull request by mikkelu**
Fixes #1804 

Added the following properties to `Abort` so all inherited properties from `Debuggable` are accessible:
- possibleCauses
- suggestedFixes
- documentationLinks
- stackOverflowQuestions
- gitHubIssues

Also made `stackTrace` optional so it conforms to `Debuggable` as discussed in #1804 

### Checklist

- [x] Circle CI is passing (code compiles and passes tests).
- [x] There are no breaking changes to public API.
- [x] New test cases have been added where appropriate.
- [x] All new code has been commented with doc blocks `///`.

### CODE REVIEW

File: Sources/Vapor/Error/Abort.swift [hunk 1]
```diff
@@ -32,18 +32,34 @@ public struct Abort: AbortError {
...
+
     /// Create a new `Abort`, capturing current source location info.
     public init(
         _ status: HTTPResponseStatus,
         headers: HTTPHeaders = [:],
         reason: String? = nil,
         identifier: String? = nil,
+		possibleCauses: [String] = [],
         suggestedFixes: [String] = [],
+		documentationLinks: [String] = [],
```
Comment (joscdk) [inline Sources/Vapor/Error/Abort.swift]: Indentation error?

File: Sources/Vapor/Error/Abort.swift [hunk 2]
```diff
@@ -32,18 +32,34 @@ public struct Abort: AbortError {
...
+	/// See `Debuggable`
+	public var gitHubIssues: [String]
+
     /// Create a new `Abort`, capturing current source location info.
     public init(
         _ status: HTTPResponseStatus,
         headers: HTTPHeaders = [:],
         reason: String? = nil,
         identifier: String? = nil,
+		possibleCauses: [String] = [],
```
Comment (joscdk) [inline Sources/Vapor/Error/Abort.swift]: Indentation error?

File: Sources/Vapor/Error/Abort.swift [hunk 3]
```diff
@@ -32,18 +32,34 @@ public struct Abort: AbortError {
...
         suggestedFixes: [String] = [],
+		documentationLinks: [String] = [],
+		stackOverflowQuestions: [String] = [],
```
Comment (joscdk) [inline Sources/Vapor/Error/Abort.swift]: Indentation error?

File: Sources/Vapor/Error/Abort.swift [hunk 4]
```diff
@@ -32,18 +32,34 @@ public struct Abort: AbortError {
...
+		gitHubIssues: [String] = [],
```
Comment (joscdk) [inline Sources/Vapor/Error/Abort.swift]: Indentation error?

File: Sources/Vapor/Error/Abort.swift [hunk 5]
```diff
@@ -53,7 +69,11 @@ public struct Abort: AbortError {
         self.headers = headers
         self.status = status
         self.reason = reason ?? status.reasonPhrase
+		self.possibleCauses = possibleCauses
```
Comment (joscdk) [inline Sources/Vapor/Error/Abort.swift]: Indentation error?

File: Sources/Vapor/Error/Abort.swift [hunk 6]
```diff
@@ -53,7 +69,11 @@ public struct Abort: AbortError {
...
         self.suggestedFixes = suggestedFixes
+		self.documentationLinks = documentationLinks
```
Comment (joscdk) [inline Sources/Vapor/Error/Abort.swift]: Indentation error?

File: Sources/Vapor/Error/Abort.swift [hunk 7]
```diff
@@ -53,7 +69,11 @@ public struct Abort: AbortError {
...
+		self.stackOverflowQuestions = stackOverflowQuestions
```
Comment (joscdk) [inline Sources/Vapor/Error/Abort.swift]: Indentation error?

File: Sources/Vapor/Error/Abort.swift [hunk 8]
```diff
@@ -53,7 +69,11 @@ public struct Abort: AbortError {
...
+		self.gitHubIssues = gitHubIssues
```
Comment (joscdk) [inline Sources/Vapor/Error/Abort.swift]: Indentation error?

File: Sources/Vapor/Error/Abort.swift [hunk 9]
```diff
@@ -32,18 +32,34 @@ public struct Abort: AbortError {
...
+    public var stackTrace: [String]?
+
+	/// See `Debuggable`
+	public var possibleCauses: [String]
     
     /// See `Debuggable`
     public var suggestedFixes: [String]
 
+	/// See `Debuggable`
+	public var documentationLinks: [String]
```
Comment (joscdk) [inline Sources/Vapor/Error/Abort.swift]: Indentation error?

File: Sources/Vapor/Error/Abort.swift [hunk 10]
```diff
@@ -32,18 +32,34 @@ public struct Abort: AbortError {
...
+
+	/// See `Debuggable`
+	public var stackOverflowQuestions: [String]
```
Comment (joscdk) [inline Sources/Vapor/Error/Abort.swift]: Indentation error?

File: Sources/Vapor/Error/Abort.swift [hunk 11]
```diff
@@ -32,18 +32,34 @@ public struct Abort: AbortError {
...
+
+	/// See `Debuggable`
+	public var gitHubIssues: [String]
```
Comment (joscdk) [inline Sources/Vapor/Error/Abort.swift]: Indentation error?

**Comment by penny-coin** [comment]
Hey @mikkelu, you just merged a pull request, have a coin! 

You now have 1 coins.

**Pull request merged**